src/averageAges.js

```diff
@@ -55,7 +71,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(
+    onlyWithSon
+      ? child => child.mother && child.sex === 'm'
+        && people.some(woman => woman.name === child.mother)
+      : child => child.mother
+        && people.some(woman => woman.name === child.mother)
+  );
```

Comment:
```suggestion
  const children = people.filter(child => onlyWithSon
    ? child.mother && child.sex === 'm'
      && people.some(woman => woman.name === child.mother)
    : child.mother
      && people.some(woman => woman.name === child.mother)
  );
```

Reply:
Thank you for suggestions and remarks. I guess, this is it.